cmd/incusd/daemon.go

```diff
@@ -2739,3 +2745,40 @@ func (d *Daemon) getOVS() (*ovs.VSwitch, error) {
 
 	return d.ovs, nil
 }
+
+func (d *Daemon) setupLinstor() error {
+	d.linstorMu.Lock()
+	defer d.linstorMu.Unlock()
+
+	// Clear any existing client
+	d.linstor = nil
+
+	// Get the Linstor controller connection string
+	controllerConnection := d.globalConfig.LinstorControllerConnection()
+
+	// Get the SSL certificates if needed
+	sslCACert, sslClientCert, sslClientKey := d.globalConfig.LinstorSSL()
+
+	// Get Linstor client
+	client, err := linstor.NewClient(controllerConnection, sslCACert, sslClientCert, sslClientKey)
+	if err != nil {
+		return fmt.Errorf("Failed to connect to Linstor: %w", err)
+	}
+
+	// Set the client
+	d.linstor = client
+
+	return nil
+}
+
+func (d *Daemon) getLinstor() (*linstor.Client, error) {
+	// Setup the client if it does not exist
+	if d.linstor == nil {
+		err := d.setupLinstor()
```

Comment:
I’m seeing a small race condition leading to a double execution of `setupLinstor`:
* `getLinstor` gets called until after the `d.linstor == nil` condition is checked
* `setupLinstor` gets called before `err := d.setupLinstor()` gets called
* `setupLinstor` gets called again by `err := d.setupLinstor()`.

I’d put a synchronization here…